pkg/framework/plugins/removepodsviolatingnodeaffinity/node_affinity.go

```diff
@@ -0,0 +1,120 @@
+type RemovePodsViolatingNodeAffinity struct {
+	handle    framework.Handle
+	args      *componentconfig.RemovePodsViolatingNodeAffinityArgs
+	podFilter podutil.FilterFunc
+}
+
+var _ framework.Plugin = &RemovePodsViolatingNodeAffinity{}
+var _ framework.DeschedulePlugin = &RemovePodsViolatingNodeAffinity{}
+
+// New builds plugin from its arguments while passing a handle
+func New(args runtime.Object, handle framework.Handle) (framework.Plugin, error) {
+	nodeAffinityArgs, ok := args.(*componentconfig.RemovePodsViolatingNodeAffinityArgs)
+	if !ok {
+		return nil, fmt.Errorf("want args to be of type RemovePodsViolatingNodeAffinityArgs, got %T", args)
+	}
+
+	var includedNamespaces, excludedNamespaces sets.String
+	if nodeAffinityArgs.Namespaces != nil {
+		includedNamespaces = sets.NewString(nodeAffinityArgs.Namespaces.Include...)
+		excludedNamespaces = sets.NewString(nodeAffinityArgs.Namespaces.Exclude...)
+	}
+
+	podFilter, err := podutil.NewOptions().
+		WithFilter(handle.Evictor().Filter).
+		WithNamespaces(includedNamespaces).
+		WithoutNamespaces(excludedNamespaces).
+		WithLabelSelector(nodeAffinityArgs.LabelSelector).
+		BuildFilterFunc()
+	if err != nil {
+		return nil, fmt.Errorf("error initializing pod filter function: %v", err)
+	}
+
+	return &RemovePodsViolatingNodeAffinity{
+		handle:    handle,
+		podFilter: podFilter,
+		args:      nodeAffinityArgs,
+	}, nil
+}
+
+// Name retrieves the plugin name
+func (d *RemovePodsViolatingNodeAffinity) Name() string {
+	return PluginName
+}
+
+func (d *RemovePodsViolatingNodeAffinity) Deschedule(ctx context.Context, nodes []*v1.Node) *framework.Status {
+	for _, nodeAffinity := range d.args.NodeAffinityType {
+		klog.V(2).InfoS("Executing for nodeAffinityType", "nodeAffinity", nodeAffinity)
+
+		switch nodeAffinity {
+		case "requiredDuringSchedulingIgnoredDuringExecution":
+		loop:
+			for _, node := range nodes {
+				klog.V(1).InfoS("Processing node", "node", klog.KObj(node))
+
+				pods, err := podutil.ListPodsOnANode(
+					node.Name,
+					d.handle.GetPodsAssignedToNodeFunc(),
+					podutil.WrapFilterFuncs(d.podFilter, func(pod *v1.Pod) bool {
+						return d.handle.Evictor().Filter(pod) &&
+							!nodeutil.PodFitsCurrentNode(d.handle.GetPodsAssignedToNodeFunc(), pod, node) &&
+							nodeutil.PodFitsAnyNode(d.handle.GetPodsAssignedToNodeFunc(), pod, nodes)
+					}),
+				)
+				if err != nil {
+					return &framework.Status{
+						Err: fmt.Errorf("error listing pods on a node: %v", err),
+					}
+				}
+
+				for _, pod := range pods {
+					if pod.Spec.Affinity != nil && pod.Spec.Affinity.NodeAffinity != nil && pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution != nil {
+						klog.V(1).InfoS("Evicting pod", "pod", klog.KObj(pod))
+						d.handle.Evictor().Evict(ctx, pod, evictions.EvictOptions{})
+						if d.handle.Evictor().NodeLimitExceeded(node) {
+							continue loop
```

Comment:
Should it not just break and move on to the next node? Seems like this is completely skipping the iteration of nodes

Reply:
The `loop` label refers to `for _, node := range nodes` loop. All good here.

Reply:
> Should it not just break and move on to the next node? Seems like this is completely skipping the iteration of nodes

@a7i I get it now :) The code got updated removing the continue in favor of the simple break.